From f4b53426f73f2d1692c4fcf89fcb42e668d6c78e Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 7 Nov 2013 05:34:35 +0530
Subject: Prevents the player to notify the client after disconnecting

BZ: 59148

The method MediaPlayerService::Client::disconnect needs to set the
notification callback prior to clearing the mClient attribute.
This is to prevent the situation when the notification occurs during
the disconnection.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: internal
upstream-candidate: yes

Change-Id: I9ab530639ed2ab5bfb749aab7137950da8cace83
Orig-Change-Id: I3b0d824a3e62e2b5d944da364db4222527df414e
Signed-off-by: Jean-Christophe PINCE <jean-christophe.pince@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libmediaplayerservice/MediaPlayerService.cpp |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/media/libmediaplayerservice/MediaPlayerService.cpp b/media/libmediaplayerservice/MediaPlayerService.cpp
index d368a9f..0733d6c 100644
--- a/media/libmediaplayerservice/MediaPlayerService.cpp
+++ b/media/libmediaplayerservice/MediaPlayerService.cpp
@@ -540,13 +540,15 @@ void MediaPlayerService::Client::disconnect()
         mClient.clear();
     }
 
-    mPlayer.clear();
-
     // clear the notification to prevent callbacks to dead client
     // and reset the player. We assume the player will serialize
     // access to itself if necessary.
-    if (p != 0) {
+    if (p != 0)
         p->setNotifyCallback(0, 0);
+
+    mPlayer.clear();
+
+    if (p != 0) {
 #if CALLBACK_ANTAGONIZER
         ALOGD("kill Antagonizer");
         mAntagonizer->kill();
-- 
1.7.4.1

